### DISCUSSION TITLE
JavaFunctions in Symja 2.0.0

### DISCUSSION THREAD

**Discussion by wirker** [axkr/symja_android_library#542] [Q&A] [answered]
Hello Axel,

according to the docs the builtin JavaFunctions (like the J/Link function in Mathematica) seem to be supported in Version 2.0.0 of Symja when using a local MMA console.

I could not, however, get it to work. `JavaNew["java.lang.String"]` evaluates to itself. `LoadJavaClass["java.lang.String"]` brings up an exception:

`java.lang.ClassCastException: class jdk.internal.loader.ClassLoaders$AppClassLoader cannot be cast to class java.net.URLClassLoader (jdk.internal.loader.ClassLoaders$AppClassLoader and java.net.URLClassLoader are in module java.base of loader 'bootstrap')`

Am I miss something like `InstallJava[]` or explicitly set `AddToClassPath`?

Best,

Stefan.

**Comment by axkr** [comment, marked as answer]
Fixed the issue in #543.

The setup needs to enable filesystem. If you want to have exact Mathematica syntax `relaxedSyntax = false` should be set:
```
        Config.FILESYSTEM_ENABLED = true; 
        boolean relaxedSyntax = false;
```

See refactored JUnit test:
https://github.com/axkr/symja_android_library/blob/a59841c81e44485850bf86e006291b2fe2f46f15/symja_android_library/matheclipse-core/src/test/java/org/matheclipse/core/system/JavaLinkTestCase.java#L89
